superpmi-collect failing for smoke_tests with error MSB4184: The expression "[System.IO.Directory]::GetDirectories(/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests)" cannot be evaluated

See for example https://dev.azure.com/dnceng/internal/_build/results?buildId=2298235&view=results

"Send job to Helix" fails with

```scala
2023-10-22T20:22:38.5505223Z ========================== Starting Command Output ===========================
2023-10-22T20:22:38.5524303Z [command]/usr/bin/bash --noprofile --norc /mnt/vss/_work/_temp/54769a59-5346-4872-ba95-62435e4d7b32.sh
2023-10-22T20:22:41.7190221Z /mnt/vss/_work/1/s/src/coreclr/scripts/superpmi-collect.proj(183,27): error MSB4184: The expression "[System.IO.Directory]::GetDirectories(/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests)" cannot be evaluated. Could not find a part of the path '/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests'.
2023-10-22T20:22:41.7258152Z ##[error]src/coreclr/scripts/superpmi-collect.proj(183,27): error MSB4184: The expression "[System.IO.Directory]::GetDirectories(/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests)" cannot be evaluated. Could not find a part of the path '/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests'.
2023-10-22T20:22:41.7267755Z 
2023-10-22T20:22:41.7268010Z Build FAILED.
2023-10-22T20:22:41.7268127Z 
2023-10-22T20:22:41.7271052Z /mnt/vss/_work/1/s/src/coreclr/scripts/superpmi-collect.proj(183,27): error MSB4184: The expression "[System.IO.Directory]::GetDirectories(/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests)" cannot be evaluated. Could not find a part of the path '/mnt/vss/_work/1/s/payload/workitem/collectAssembliesDirectory/smoke_tests'.
2023-10-22T20:22:41.7273739Z     0 Warning(s)
2023-10-22T20:22:41.7273962Z     1 Error(s)
2023-10-22T20:22:41.7274074Z 
2023-10-22T20:22:41.7274270Z Time Elapsed 00:00:01.88
2023-10-22T20:22:41.7495746Z Build failed with exit code 1. Check errors above.
2023-10-22T20:22:41.7620205Z ##[section]Finishing: Send job to Helix (Unix)
```

cc @dotnet/jit-contrib @TIHan can you take a look?

--- a/src/coreclr/scripts/superpmi_collect_setup.py
+++ b/src/coreclr/scripts/superpmi_collect_setup.py
@@ -561,7 +561,9 @@ def main(main_args):
 
         # Build nativeaot tests
         if coreclr_args.collection_type == "nativeaot":
+            build_file = "build.cmd" if is_windows else "build.sh"
             tests_build_file = "build.cmd" if is_windows else "build.sh"
+            run_command([os.path.join(source_directory, build_file), "Tools.ILLink", "-arch", arch, "-c", coreclr_args.build_type])
             run_command([os.path.join(tests_directory, tests_build_file), "nativeaot", arch, coreclr_args.build_type, "tree", "nativeaot/SmokeTests"], source_directory)
 
         # NOTE: we can't use the build machine ".dotnet" to run on all platforms. E.g., the Windows x86 build uses a

